glean-core/src/histogram/mod.rs

```diff
@@ -0,0 +1,83 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+//! A simple histogram implementation for exponential histograms.
+
+use std::collections::HashMap;
+
+use serde::{Deserialize, Serialize};
+
+pub use exponential::Exponential;
+pub use functional::Functional;
+
+mod exponential;
+mod functional;
+
+/// A histogram.
+///
+/// Stores the ranges of buckets as well as counts per buckets.
+/// It tracks the count of added values and the total sum.
+///
+/// ## Example
+///
+/// ```rust,ignore
+/// let mut hist = Histogram::exponential(1, 500, 10);
+///
+/// for i in 1..=10 {
+///     hist.accumulate(i);
+/// }
+///
+/// assert_eq!(10, hist.count());
+/// assert_eq!(55, hist.sum());
+/// ```
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct Histogram<B> {
+    // bucket minimum -> value
+    values: HashMap<u64, u64>,
+
+    count: u64,
+    sum: u64,
+
+    bucketing: B,
+}
+
+pub trait Bucketing {
```

Comment:
NICE! While you already mention that documenting traits it's uncommon in the Rust community, I'd really appreciate if we could document what is this trait for :o

Reply:
I never said documenting traits is uncommon! 

This absolutely needs documentation.

Reply:
> 
> 
> I never said documenting traits is uncommon!

My memory is definitely failing me these days. Sorry for putting words in your "fingers" :-)